pkg/config/config.go

```diff
@@ -205,34 +230,76 @@ var (
 	hasCustomAlertmanagerConfig = false
 	certDuration                = time.Hour * 24 * 365
 
-	Replicas1      int32 = 1
-	Replicas2      int32 = 2
-	Replicas3      int32 = 3
-	thanosReplicas       = map[string]*int32{
+	Replicas1 int32 = 1
+	Replicas2 int32 = 2
+	Replicas3 int32 = 3
+	Replicas        = map[string]*int32{
 		ObservatoriumAPI:    &Replicas2,
 		ThanosQuery:         &Replicas2,
 		ThanosQueryFrontend: &Replicas2,
 		Grafana:             &Replicas2,
-		RbacQueryProxy:      &Replicas2,
-
-		ThanosRule:           &Replicas3,
-		ThanosReceive:        &Replicas3,
-		ThanosStoreMemcached: &Replicas3,
-		Alertmanager:         &Replicas3,
+		RBACQueryProxy:      &Replicas2,
+
+		ThanosRule:                   &Replicas3,
+		ThanosReceive:                &Replicas3,
+		ThanosStoreShard:             &Replicas3,
+		ThanosStoreMemcached:         &Replicas3,
+		ThanosQueryFrontendMemcached: &Replicas3,
+		Alertmanager:                 &Replicas3,
 	}
-)
 
-func GetObservabilityComponentReplicas(componentName string) *int32 {
-	return thanosReplicas[componentName]
-}
+	MemoryLimitMB   = int32(1024)
+	ConnectionLimit = int32(1024)
+	MaxItemSize     = "1m"
+)
 
-func SetObservabilityComponentReplicas(name string, replicas *int32) {
-	for k := range thanosReplicas {
-		if name == GetMonitoringCRName()+"-"+k {
-			thanosReplicas[k] = replicas
-			return
+func GetReplicas(component string, advanced *observabilityv1beta2.AdvancedConfig) *int32 {
+	if advanced == nil {
+		return Replicas[component]
+	}
+	var replicas *int32
+	switch component {
+	case ObservatoriumAPI:
+		if advanced.ObservatoriumAPI != nil {
+			replicas = advanced.ObservatoriumAPI.Replicas
+		}
+	case ThanosQuery:
+		if advanced.Query != nil {
+			replicas = advanced.Query.Replicas
+		}
+	case ThanosQueryFrontend:
+		if advanced.QueryFrontend != nil {
+			replicas = advanced.QueryFrontend.Replicas
+		}
+	case ThanosQueryFrontendMemcached:
+		if advanced.QueryFrontendMemcached != nil {
+			replicas = advanced.QueryFrontendMemcached.Replicas
```

Comment:
we may need to figure out how to eliminate the switch sentence, but I can't find a better solution.
Maybe we can change `AdvancedConfig` go struct in API to something like `map[string]ComponentSetting`, but that will losing validation...

Reply:
Need more thinking about it. maybe have another PR to enhancement later.